### PR TITLE
Update dependencies eslint config package in client packages

### PR DESCRIPTION
**Pull request by Josmithr**
Also fixes linter violtions.

Linter config was updated in other release groups in #12128.

### CODE REVIEW

**Review by sonalivdeshpande — APPROVED** [review]
looks good :shipit:

File: examples/apps/collaborative-textarea/package.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
   "devDependencies": {
     "@fluid-tools/webpack-fluid-loader": ">=2.0.0-internal.1.2.0 <2.0.0-internal.2.0.0",
     "@fluidframework/build-common": "^1.0.0",
-    "@fluidframework/eslint-config-fluid": "^0.28.2000",
+    "@fluidframework/eslint-config-fluid": "^1.0.0",
     "@fluidframework/test-tools": "^0.2.3074",
     "@fluidframework/test-utils": ">=2.0.0-internal.1.2.0 <2.0.0-internal.2.0.0",
     "@rushstack/eslint-config": "^2.5.1",
```
Comment (tylerbutler) [inline examples/apps/collaborative-textarea/package.json:59]: Nit: is this still needed?

**Review by tylerbutler — APPROVED** [review]

**Comment by tylerbutler** [comment]
@sonalivdeshpande any idea why the bundle size calculation still doesn't have a baseline in next?

File: examples/apps/collaborative-textarea/package.json (hunk 1 above)
Comment (Josmithr) [reply examples/apps/collaborative-textarea/package.json:59]: Sorry, can you elaborate?

File: packages/dds/merge-tree/src/test/client.applyMsg.spec.ts [hunk 2]
```diff
@@ -530,59 +530,63 @@ describe("client.applyMsg", () => {
...
-           |               |            |          b    |            |          c    |            |
-7:6:B0@6   | DDDDDDb ---D  | 7:6:B0@6   | DDDDDD- -b-D  | 7:6:B0@6   | DDDDDDb- -_-D | 7:6:B0@6   | DDDDDDb ---D
-           |               |            |               |            |           c   |            |
-8:6:C0@6   | DDDDDDcb ---D | 8:6:C0@6   | DDDDDDc- -b-D | 8:6:C0@6   | DDDDDDb- -c-D | 8:6:C0@6   | DDDDDDcb ---D
-Client C does not match client A
-```
+     * ```
+     * _: Local State
+     * -: Deleted
+     *: Unacked Insert and Delete
```
[the remark this answers is not in the document]
Comment (Josmithr) [reply packages/dds/merge-tree/src/test/client.applyMsg.spec.ts:538]: Good catch! Will fix.

File: examples/apps/collaborative-textarea/package.json (hunk 1 above)
Comment (Josmithr) [reply examples/apps/collaborative-textarea/package.json:59]: Oh, github was highlighting the wrong line. No, this shouldn't be needed. Will remove.

Comment (Josmithr) [reply examples/apps/collaborative-textarea/package.json]: I also just put up a PR on main that removes this dep globally: #12144

Comment (Josmithr) [reply examples/apps/collaborative-textarea/package.json]: Hmm. Actually, in most packages this seems to cause problems. The config itself isn't needed, but the plugins that normally get pulled in with it are. I suspect we could work around this by using the inline require pattern in our base config, but that would need to be validated.

**Pull request merged**